modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/UserAgentProcessor.java

```diff
@@ -236,13 +310,31 @@ public UserAgentProcessor create(Map<String, Processor.Factory> factories, Strin
                 properties = EnumSet.allOf(Property.class);
             }
 
-            return new UserAgentProcessor(processorTag, field, targetField, parser, properties, ignoreMissing);
+            if (useECS == false) {
+                deprecationLogger.deprecated("setting [ecs] to false for non-common schema " +
+                    "format is deprecated and will be removed in 7.0, set to true to use the non-deprecated format");
+            }
+
+            return new UserAgentProcessor(processorTag, field, targetField, parser, properties, ignoreMissing, useECS);
         }
     }
 
     enum Property {
 
-        NAME, MAJOR, MINOR, PATCH, OS, OS_NAME, OS_MAJOR, OS_MINOR, DEVICE, BUILD;
+        NAME,
+        // Deprecated in 6.7 (superceded by VERSION), to be removed in 7.0
+        @Deprecated MAJOR,
```

Comment:
should we also log deprecation warning if a deprecated property is used?

Reply:
Sounds good, I added this, thanks for the reminder